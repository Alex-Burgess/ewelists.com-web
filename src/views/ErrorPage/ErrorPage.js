import React from "react";
import { Link } from "react-router-dom";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
// @material-ui/icons
// core components
import HeaderFixed from "components/Header/HeaderFixed.js";
import Footer from "components/Footer/FooterGrey.js";
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";

import styles from "assets/jss/material-kit-pro-react/views/errorPageStyle.js";
const useStyles = makeStyles(styles);

export default function ErrorPage(props) {
  const classes = useStyles();
  return (
    <div className={classes.page}>
      <HeaderFixed isAuthenticated={props.isAuthenticated} user={props.user} mobile={props.mobile} />
      <div className={classes.container}>
        <GridContainer justify="center">
          <GridItem xs={12} sm={10} md={7}>
            <h1 className={classes.title}>Oops!</h1>
            <h2 className={classes.subTitle}>We can't seem to find the page you're looking for.</h2>
            <div>
              You could head back to the
              <Link to="/"> home page</Link>
              , or if this keeps happening,
              <Link to="/contact"> drop us a line </Link>
               so we can quickly fix it.
            </div>

          </GridItem>
        </GridContainer>
      </div>
      <div className={classes.flexer} />
      <Footer />
    </div>
  );
}
