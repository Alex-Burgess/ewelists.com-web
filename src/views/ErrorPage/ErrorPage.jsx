/*!

=========================================================
* Material Kit PRO React - v1.7.0
=========================================================

* Product Page: https://www.creative-tim.com/product/material-kit-pro-react
* Copyright 2019 Creative Tim (https://www.creative-tim.com)

* Coded by Creative Tim

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
/*eslint-disable*/
import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import withStyles from "@material-ui/core/styles/withStyles";
// @material-ui/icons
// core components
import Header from "components/Header/Header.jsx";
import HeaderLinks from "components/Header/HeaderLinks.jsx";
import FooterTransparent from "components/Footer/FooterTransparent.jsx";
import GridContainer from "components/Grid/GridContainer.jsx";
import GridItem from "components/Grid/GridItem.jsx";

import errorPageStyle from "assets/jss/material-kit-pro-react/views/errorPageStyles.jsx";

import image from "assets/img/sheep-with-shoes.jpg";

class ErrorPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      checked: [1]
    };
    this.handleToggle = this.handleToggle.bind(this);
  }
  handleToggle(value) {
    const { checked } = this.state;
    const currentIndex = checked.indexOf(value);
    const newChecked = [...checked];

    if (currentIndex === -1) {
      newChecked.push(value);
    } else {
      newChecked.splice(currentIndex, 1);
    }

    this.setState({
      checked: newChecked
    });
  }
  componentDidMount() {
    window.scrollTo(0, 0);
    document.body.scrollTop = 0;
  }
  render() {
    const { classes, ...rest } = this.props;
    return (
      <div>
        <Header
          absolute
          color="transparent"
          brand="ewelist"
          links={<HeaderLinks dropdownHoverColor="dark" />}
          {...rest}
        />
        <div
          className={classes.pageHeader}
          style={{
            backgroundImage: "url(" + image + ")",
            backgroundSize: "cover",
            backgroundPosition: "top center"
          }}
        >
          <div className={classes.contentCenter}>
            <GridContainer>
              <GridItem md={12}>
                <h1 className={classes.title}>Page not found</h1>
                <h2 className={classes.subTitle}>The page you are looking for does not exist.</h2>
                <a href="/">Click to go back to the home page.</a>
              </GridItem>
            </GridContainer>
          </div>
        </div>

        <FooterTransparent />
      </div>
    );
  }
}

ErrorPage.propTypes = {
  classes: PropTypes.object
};

export default withStyles(errorPageStyle)(ErrorPage);
