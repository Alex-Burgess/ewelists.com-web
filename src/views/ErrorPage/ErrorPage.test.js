import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ErrorPage from './ErrorPage';

window.matchMedia = jest.fn().mockImplementation(query => {
  return {
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  };
});

test('renders error page', () => {
  const tree = renderer
    .create(<Router><ErrorPage /></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
