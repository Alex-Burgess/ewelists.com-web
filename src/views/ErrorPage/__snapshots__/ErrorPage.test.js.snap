// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error page 1`] = `
<div>
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed Header-appBar-14 Header-transparent-26 Header-absolute-15 MuiAppBar-colorPrimary mui-fixed"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular Header-container-17 MuiToolbar-gutters"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="Header-logo-33"
          src="logo-white.png"
        />
      </a>
      <div
        className="Header-hidden-30 PrivateHiddenCss-smDown-79"
      >
        <div
          className="Header-collapse-31"
        >
          <ul
            className="MuiList-root HeaderLinks-list-89 HeaderLinks-mlAuto-105 MuiList-padding"
          >
            <li
              className="MuiListItem-root HeaderLinks-listItem-90 MuiListItem-gutters"
              disabled={false}
            >
              <a
                className="HeaderLinks-navLink-92"
                href="/listideas"
                onClick={[Function]}
              >
                Gift List Ideas
              </a>
            </li>
            <li
              className="MuiListItem-root HeaderLinks-listItem-90 MuiListItem-gutters"
              disabled={false}
            >
              <a
                className="HeaderLinks-navLink-92"
                href="/login"
                onClick={[Function]}
              >
                Log In
              </a>
            </li>
            <li
              className="MuiListItem-root HeaderLinks-listItem-90 MuiListItem-gutters"
              disabled={false}
            >
              <a
                className="HeaderLinks-navLink-92"
                href="/signup"
                onClick={[Function]}
              >
                Sign Up
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>
  <div
    className="ErrorPage-pageHeader-5"
    style={
      Object {
        "backgroundImage": "url(sheep-with-shoes.jpg)",
        "backgroundPosition": "top center",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="ErrorPage-contentCenter-1"
    >
      <div
        className="MuiGrid-root GridContainer-grid-121  MuiGrid-container"
      >
        <div
          className="MuiGrid-root GridItem-grid-225  MuiGrid-item MuiGrid-grid-md-12"
        >
          <h1
            className="ErrorPage-title-2"
          >
            Page not found
          </h1>
          <h2
            className="ErrorPage-subTitle-3"
          >
            The page you are looking for does not exist.
          </h2>
          <a
            href="/"
          >
            Click to go back to the home page.
          </a>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="Footer-footer-229 ErrorPage-footer-13"
  >
    <div
      className="Footer-container-235"
    >
       
      <div>
        <div
          className="ErrorPage-left-9"
        >
          <ul
            className="MuiList-root ErrorPage-list-8 MuiList-padding"
          >
            <li
              className="MuiListItem-root ErrorPage-inlineBlock-7 MuiListItem-gutters"
              disabled={false}
            >
              <a
                className="ErrorPage-block-6"
                href="/"
              >
                Â© 
                2019
                 ewelists
              </a>
            </li>
            <li
              className="MuiListItem-root ErrorPage-inlineBlock-7 MuiListItem-gutters"
              disabled={false}
            >
              <a
                className="ErrorPage-block-6"
                href="/contact"
              >
                Contact Us
              </a>
            </li>
            <li
              className="MuiListItem-root ErrorPage-inlineBlock-7 MuiListItem-gutters"
              disabled={false}
            >
              <a
                className="ErrorPage-block-6"
                href="/#about"
              >
                About us
              </a>
            </li>
            <li
              className="MuiListItem-root ErrorPage-inlineBlock-7 MuiListItem-gutters"
              disabled={false}
            >
              <a
                className="ErrorPage-block-6"
                href="/privacy"
              >
                Privacy
              </a>
            </li>
            <li
              className="MuiListItem-root ErrorPage-inlineBlock-7 MuiListItem-gutters"
              disabled={false}
            >
              <a
                className="ErrorPage-block-6"
                href="/terms"
              >
                Terms
              </a>
            </li>
          </ul>
        </div>
        <div
          className="ErrorPage-rightLinks-11"
        >
          <ul>
            <li>
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiButton-root ForwardRef-button-246 ForwardRef-simple-269 ForwardRef-justIcon-277 MuiButton-text"
                href="https://www.facebook.com/ewelists"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
                target="_blank"
              >
                <span
                  className="MuiButton-label"
                >
                  <i
                    className="fab fa-facebook"
                  />
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </a>
            </li>
            <li>
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiButton-root ForwardRef-button-246 ForwardRef-simple-269 ForwardRef-justIcon-277 MuiButton-text"
                href="https://twitter.com/ewelists"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
                target="_blank"
              >
                <span
                  className="MuiButton-label"
                >
                  <i
                    className="fab fa-twitter"
                  />
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </a>
            </li>
            <li>
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiButton-root ForwardRef-button-246 ForwardRef-simple-269 ForwardRef-justIcon-277 MuiButton-text"
                href="https://www.instagram.com/ewelists/"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
                target="_blank"
              >
                <span
                  className="MuiButton-label"
                >
                  <i
                    className="fab fa-instagram"
                  />
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        className="Footer-clearFix-245"
      />
    </div>
  </footer>
</div>
`;
